lib/matplotlib/backend_tools.py

```diff
@@ -318,7 +321,12 @@ class ToolQuit(ToolBase):
     default_keymap = rcParams['keymap.quit']
 
     def trigger(self, sender, event, data=None):
-        Gcf.destroy_fig(self.figure)
+        try:
+            manager = self.figure.canvas.manager
+        except:
```

Comment:
Expecting to only get an AttributeError is no excuse for a bare exception. At the very least, it makes it self-documenting, and at best, it future-proofs the code. Silently swallowing bare exceptions is a huge no-no.